src/main/kotlin/com/procurify/flagcounter/FlagEquivalentMessageGenerator.kt

```diff
@@ -13,20 +13,21 @@ object FlagEquivalentMessageGenerator {
         // Convert the number of binary flags into the corresponding base ten power
         val baseTenExponent = (flagCount * log10(2.0)).toInt()
 
-        return numberMap[baseTenExponent]?.let {
-            "That's as many configurations as $it!"
-        } ?: findTwoSum(baseTenExponent).let {
-            if (it.size == 2) {
-                val firstString = numberMap[it[0]]
-                val secondString = numberMap[it[1]]
+        val singleStringOrNull by lazy { numberMap[baseTenExponent] }
+        val twoSumOrEmpty by lazy { findTwoSum(baseTenExponent) }
+
+        return when {
+            singleStringOrNull != null -> "That's as many configurations as $singleStringOrNull!"
+            twoSumOrEmpty.size == 2 -> {
+                val firstString = numberMap[twoSumOrEmpty[0]]
+                val secondString = numberMap[twoSumOrEmpty[1]]
                 if (firstString != null && secondString != null) {
                     "That's as many configurations as $firstString multiplied by $secondString!"
                 } else {
-                    null
+                    "That's 10^$baseTenExponent different configurations!"
```

Comment:
Would there be a way to roll this into the lower `else` condition as well? Perhaps by adding an "or" condition to the `twoSumOrEmpty.size == 2` portion of the `when`?

Reply:
Good call, I wasn't satisfied with this. I played around with that or condition but I think it ended up less readable. 

I think I might just pull the `firstString` and `secondString` up with better names and add the or that way